util/config.py

```diff
@@ -120,14 +120,15 @@ def initialize_globals():
         with tf.device('/job:ps/task:%d' % i):
             done_queues.append(tf.FIFOQueue(1, tf.int32, shared_name=('queue%i' % i)))
 
-    # Placeholder to pass in the worker's index as token
-    c.token_placeholder = tf.placeholder(tf.int32)
+    if not tf.executing_eagerly():
```

Comment:
Here to allow testing in eager mode more easily.

Reply:
What are the reasons for not moving to eager execution entirely?
I'd even vote for ditching MonitoreTrainingSession centric code and the coordinator.

Reply:
Mostly to keep changes smaller so that they can be merged more easily, but also because moving to Eager is a much more involved task, we can't just flip the switch right now. The training loop has to be restructured significantly. I plan to work on that in the near/medium term.